pkg/eval/external_cmd_test.go

```diff
@@ -0,0 +1,41 @@
+package eval
+
+import (
+	"os"
+	"testing"
+)
+
+func TestBuiltinFnExternal(t *testing.T) {
+	tmpHome, cleanup := InTempHome()
+	defer cleanup()
+
+	mustMkdirAll("dir")
+	mustCreateEmpty("cmdx")
+	mustWriteFile("cmd1", []byte("#!/bin/sh"), 0755)
+	mustWriteFile("cmd2", []byte("#!/bin/sh"), 0755)
+	mustWriteFile("cmd3", []byte(""), 0755)
+	mustCreateEmpty("file")
+
+	os.Setenv("PATH", tmpHome+":"+os.Getenv("PATH"))
+	Test(t,
+		That(`resolve external`).Puts("$external~"),
+		That(`e = (external true); kind-of $e`).Puts("fn"),
+		That(`e = (external true); put (repr $e)`).Puts("<external true>"),
+		That(`e = (external false); m = [&$e=true]; put (repr $m)`).Puts("[&<external false>=true]"),
+		// This group tests the `ExternalCmd.Call` method.
+		That(`e = (external true); $e`).DoesNothing(),
+		That(`e = (external true); $e &option`).Throws(ErrExternalCmdOpts, "$e &option"),
+		That(`e = (external false); $e`).ThrowsCmdExit(
+			ExternalCmdExit{CmdName: "false", WaitStatus: ExitWaitStatus(1)}),
+		// TODO: Modify the ExternalCmd.Call method to wrap the Go error in a
+		// predictable Elvish error so we don't have to resort to using
+		// ThrowsAny in the following tests.
+		//
+		// The command shouldn't be found when run so we should get an
+		// exception along the lines of "executable file not found in $PATH".
+		That(`e = (external true); E:PATH=/; $e`).ThrowsAny(),
```

Comment:
Don't mutate E:PATH; use a temporary assignment.

Reply:
Done. I wrote it this way because I hadn't decided to un-deprecate, and test, the `$pwd` behavior when I wrote this. Since you seem to be willing to retain `$pwd`, at least for the time being, I've modified this test to use it.